woofbot-lnd/docker-compose.yml

```diff
@@ -0,0 +1,42 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: woofbot-lnd_web_1
+      APP_PORT: 8080
+
+  web:
+    image: woofbot/woofbot:v0.7.0@sha256:36f024668d4c27738f3e2676bc949408b63e2bdffd2c2e450c8c0ae1f0776956
+    restart: on-failure
+    stop_grace_period: 1m
+    environment:
+      MONGODB_URI: "mongodb://woofbot-lnd_mongodb_1:27017/woofbot"
+      APP_PORT: "8080"
+      APP_SEED: $APP_SEED
+      APP_BITCOIN_NODE_IP: $APP_BITCOIN_NODE_IP
+      APP_BITCOIN_RPC_USER: $APP_BITCOIN_RPC_USER
+      APP_BITCOIN_RPC_PASS: $APP_BITCOIN_RPC_PASS
+      APP_BITCOIN_RPC_PORT: $APP_BITCOIN_RPC_PORT
+      APP_LIGHTNING_NODE_IP: $APP_LIGHTNING_NODE_IP
+      APP_LIGHTNING_NODE_GRPC_PORT: $APP_LIGHTNING_NODE_GRPC_PORT
+      LND_TLS_PATH: "/lnd/tls.cert"
+      LND_READONLY_MACAROON_PATH: "/lnd/data/chain/bitcoin/$APP_BITCOIN_NETWORK/readonly.macaroon"
+      SERVER_LOGS_FILEPATH: /app/logs/server.log
+    volumes:
+      - ${APP_DATA_DIR}/data/server-logs:/app/logs
+      - "${APP_LIGHTNING_NODE_DATA_DIR}/tls.cert:/lnd/tls.cert:ro"
+      - "${APP_LIGHTNING_NODE_DATA_DIR}/data/chain/bitcoin/${APP_BITCOIN_NETWORK}/readonly.macaroon:/lnd/data/chain/bitcoin/${APP_BITCOIN_NETWORK}/readonly.macaroon:ro"
```

Comment:
Here you can just mount the entire LND data directory as read only. Like this:
`${APP_LIGHTNING_NODE_DATA_DIR}:/lnd:ro`

What you have done makes perfect sense in the context of sharing as little file system access as possible for increased isolation. But in this specific case since the TLS cert is generated dynamically by LND there is an extreme edge-case where WoofBot starts before LND has created the cert. In this case, `${APP_LIGHTNING_NODE_DATA_DIR}/tls.cert` path won't exist when the container is starting so Docker will create an empty directory at that path owned by root. This could lead to LND running into issues when it's trying to create it's TLS cert.

Reply:
@nmfretz Indeed WoofBot needs only readonly access to LND.

Is there no other way? What if the docker will deliberately crash if LND_TLS_PATH is defined but the file was not found?
Or maybe even have a small separated service that will mount `/lnd` and then copy the tls.cert and readonly.macaroon to the APP_DATA_DIR folder?

Reply:
@nmfretz I see that ln-visualizer is doing something similar:

https://github.com/getumbrel/umbrel-apps/blob/master/ln-visualizer/docker-compose.yml#L24-L25

Maybe adding `user: 1000:1000` would solve the issue of the directory won't be created as root and will be mounted when LND starts?

Reply:
> What if the docker will deliberately crash if LND_TLS_PATH is defined but the file was not found?

Ah, good point. Yes, let's try `user 1000:1000` in the web container and see if the permissions allow your app to write properly.

Reply:
I just tested with `user: "1000:1000"` and everything works as expected.

Reply:
Great, I updated the PR to have `user: 1000:1000`.
I'm a little bit confused whether the pre-start hook is still necessary. The relevant discussion about it can be found here:
https://github.com/getumbrel/umbrel-apps/issues/346
https://github.com/getumbrel/umbrel-apps/pull/353#pullrequestreview-1236142421

It doesn't hurt to have this script create and chown the `data/server-logs` folder, so I'll leave it as-is for now.

I'm also trying now to uninstall and reinstall the app, and will let you know if it works.

Reply:
Ya, I read through these the other day and let's leave the pre-start hook for now because the next version of Umbrel that includes @nevets963's fix for this issue hasn't been released yet.

Reply:
Everything works smoothly. I think we are good to go.

Reply:
Excellent!